### ISSUE TITLE
"arguments" incorrectly munged to "arguments_1" in numerous circumstances in a for loop targeting ES5

### ISSUE THREAD

**Issue by Taytay** [microsoft/TypeScript#38594]
**TypeScript Version:**  Nightly (also, 3.9.2, 3.7.4, and probably others)

**Search Terms:**
arguments property, arguments_1 renamed

**Expected behavior:**
When targeting ES5, an object declared like so:
`{arguments: "foo"}` 
is emitted to javascript as:
`{arguments: "foo"}`

**Actual behavior:**
The object is emitted to javascript as: 
`{arguments_1: "foo"}`

See below code sample for more related bugs.

**Related Issues:**

A similar variant was reported and correctly fixed here:
https://github.com/microsoft/TypeScript/issues/13586

This exact bug was previously reported here, and closed as "fixed" by the reporter. (I don't think it was actually fixed though). 
https://github.com/microsoft/TypeScript/issues/33540

**Code**
```ts
// Compile with ES5 target
function myFunc(_param: string) {
    for (let x = 0; x < 1; ++x){
        let i : number;
        // It's essential to have an inner function declared that references a variable defined in the loop
        [].forEach(function () { i })

        let brokenObj1 = {
            arguments: 0
        }
        // Actual:   {"arguments_1":0}
        // Expected: {"arguments":0}
        console.log(JSON.stringify(brokenObj1)); 

        let brokenObj2 = {
            arguments
        }
        // Actual:   {"arguments":{"0":0}}
        // Expected: {"arguments":{"0":"paramValue"}}
        // (Becuase it erroneously references the arguments of the generated loop function `_loop_1`)
        console.log(JSON.stringify(brokenObj2)); 
        

        let brokenObj3 = {
            arguments: arguments
        }
        // Actual:   {"arguments_1":{"0":"paramValue"}}
        // Expected: {"arguments":{"0":"paramValue"}}
        console.log(JSON.stringify(brokenObj3)); 
    }
}

myFunc("paramValue");
```

<details><summary><b>Output</b></summary>

```ts
"use strict";
// Compile with ES5 target
function myFunc(_param) {
    var _loop_1 = function (x) {
        var i;
        // It's essential to have an inner function declared that references a variable defined in the loop
        [].forEach(function () { i; });
        var brokenObj1 = {
            arguments_1: 0
        };
        // Actual:   {"arguments_1":0}
        // Expected: {"arguments":0}
        console.log(JSON.stringify(brokenObj1));
        var brokenObj2 = {
            arguments: arguments
        };
        // Actual:   {"arguments":{"0":0}}
        // Expected: {"arguments":{"0":"paramValue"}}
        // (Becuase it erroneously references the arguments of the generated loop function `_loop_1`)
        console.log(JSON.stringify(brokenObj2));
        var brokenObj3 = {
            arguments_1: arguments_1
        };
        // Actual:   {"arguments_1":{"0":"paramValue"}}
        // Expected: {"arguments":{"0":"paramValue"}}
        console.log(JSON.stringify(brokenObj3));
    };
    var arguments_1 = arguments;
    for (var x = 0; x < 1; ++x) {
        _loop_1(x);
    }
}
myFunc("paramValue");

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/?target=1&ts=4.0.0-dev.20200514&ssl=1&ssc=1&pln=34&pc=22#code/PTAEGEHsFsAcEsA2BTUB3eAXAFqAogMoCsomAhgE4DmymAUAGYCuAdgMabyQujQCeAMVZsAFAH1YlMtABcoAM6YK8FlQCUoAN51Qu0A0gVQIlJlAAPUAF5QABgDcF0AB5QARkcBqT+bXa9AaCmoPCgcixM0ABGyBT2OoG6IKAAkpgA5PKgyPLyyCycZIikkKDYZABuqGQ8Kiyx+sKc3KAAJshsiJTIraTlZhTIDLH5bDmgZKAVlPBkUShtQyo9ITw4qIiQkLAJiQDaALoAdAYUeGRs2CLM7M08IhqaIaAAvmp0u4HBURSQANb5ADyUQAVm5rFpPoldJQqJF8ph5HJbFC9C8PtDQMkAIIcJhFOS6TQAIlh8IK8jEbmJMls6MxyTw5lgHUwPTkJLJ0AR8hpdNRujY3HkkBQR02VBEACkCICAHJHRTKVTwBh8EQ-f5A0FuNRqRwY6HfX4AljAkEAJgh-kxei5PIFr0dOLxBL0nOo5MRNJJtj5L3p0MZzNZ7K0pM93IpPuJfpkxMkFGkADUikxkMSA86wCIAEIdfF5EJmWK-eqQJjyRB8UCDYaDdjjdYTSM80CQBh9VA0epJtm9TbbRq3Lg8AAGYkHsCpY-emKFLBFYol0tlCqVKioqvVmtN5oteoNmMNiWNWrNoIAzNbHQF7RS5PfEY7A4kXZh8YhCeGn5TqTJfRpBMpGgVNEHTTNX0CYMWQ4MMPThKNvQA2MgMTFM0wzLN52FUVkHFSBJRleVFSUTdtw1E1tRBS9D1AT50XROh+CEdgRGApNQMw4l9SAA)
     
P.S. Typescript is awesome. We ❤️Typescript!
